privacy_evaluator/attacks/property_inference_attack.py

```diff
@@ -329,51 +423,55 @@ def prediction_on_specific_property(
 
         return prediction
 
-    def attack(self):
+    def attack(self) -> Tuple[str, Dict[str, float]]:
         """
         Perform Property Inference attack.
-        :param params: Example data to run through target model for feature extraction
-        :type params: np.ndarray
-        :return: prediction about property of target data set
-            [[1, 0]]-> property; [[0, 1]]-> negation property
-        :rtype: np.ndarray with shape (1, 2)
+        :return: message with most probable property, dictionary with all properties
         """
-
+        if self.verbose > 0:
+            print("Initiating Property Inference Attack ... ")
+            print("Extracting features from target model ... ")
         # extract features of target model
         feature_extraction_target_model = self.feature_extraction(self.target_model)
 
-        # set ratio and size for unbalanced data sets
-        size_set = 1000
+        if self.verbose > 0:
+            print(
+                feature_extraction_target_model.shape,
+                " --- features extracted from the target model.",
+            )
 
         # balanced ratio
-        num_elements = int(round(size_set / 2))
-        neg_property_num_elements_per_class = {0: num_elements, 1: num_elements}
-
+        num_elements = int(round(self.size_set / len(self.classes)))
+        neg_property_num_elements_per_class = {i: num_elements for i in self.classes}
+
+        if self.verbose > 0:
+            print(
+                "Creating set of",
+                int(self.amount_sets / 2),
+                "balanced shadow classifiers ... ",
+            )
         # create balanced shadow classifiers negation property
         shadow_classifiers_neg_property = (
             self.create_shadow_classifier_from_training_set(
                 neg_property_num_elements_per_class
             )
         )
 
-        predictions = {}
+        self.ratios_for_attack.sort()
+        predictions = OrderedDict.fromkeys(self.ratios_for_attack, 0)
         # iterate over unbalanced ratios in 0.05 steps (0.05-0.45, 0.55-0.95)
```

Comment:
I think this documentation should be moved below (since it has nothing to to with the verbose, but with the lines under that)